pkgs/build-support/docker/default.nix

```diff
@@ -434,6 +434,8 @@ rec {
         if [[ -n "$fromImage" ]]; then
           echo "Unpacking base image..."
           tar -C image -xpf "$fromImage"
+          # Do not import the base image configuration
+          rm -f image/*.json
```

Comment:
Yes, the `manifest.json` should be definitely deleted. The other file is the image config of the parent image, which could be (accordingly to the spec) optionally referenced in the `manifest.json`, but as `buildImage` doesn’t (yet) produce such a file it would be also ok the delete the config of the parent.